design/2022_Q2_data_citations.md

```diff
@@ -0,0 +1,217 @@
+
+Pros:
+
+* Relatively simple implementation
+* Allows for more information about the person, including theoretically unique IDs (although
+  a person can have many KBase accounts. A person *should* only have one OrcID but it's not clear
+  if that constraint is enforced or enforceable).
+* Allows for adding more fields in the future to add more information about people in data
+  citations.
+  
+Cons:
+
+* In many cases only the person's name will be available
+* Names don't uniquely identify people
+* Names can change over time, leaving outdated records in the database
+* The same names can be represented differently, meaning the database could have different
+  records that refer to the same person
+* The name could be entered differently for the same KBase or OrcID in different records
+* If a person creates a KBase ID or OrcID after data creation, that will not be reflected in
+  the database records
+
+Estimated work: 1 week
+
+Further options:
+
+* Validate OrcIDs
+  * Needs further research but probably work estimate + 1 week
+* If a KBase ID is present, populate the name and OrcID (if present) from the Auth2 server
+  * The workspace will need access to an Auth2 Administrator account
+  * Work estimate + 1 week
+  * Note that the OrcID in the Auth2 server can change or be removed for a specific user over time,
+    hence storing them permanently with the user record at the time of creation
+
+### Names option 3 - central DB with unique ID in structure
+
+In this case the `authors` field is a list of IDs, each of which uniquely identifies a person
+in the world. Each IDs refers to a record in a database which contains the user's name,
+optionally a KBase ID, and optionally an OrcID. Other fields may be present as needed.
+
+This option requires the discovery or creation of a database of humans, uniquely
+identified with no duplication, with the fields we need. The database must be updateable by
+anyone who wishes to add EDUs to an object, and must ensure that duplicate records are not
+added for the same person.
+
+Pros:
+
+* Each ID uniquely identifies a single person in the world. Workspace provenance records
+  will not contain multiple different records referring to the same person.
+* As a user's information changes over time, those changes can be reflected in views of the
+  EDU data by pulling the information from the central database, rather than storing it in the
+  records.
+* Allows for adding more fields in the future to add more information about people in data
+  citations.
+
+Cons:
+
+* Extremely difficult implementation. It is not clear if there is any way to prevent duplicate
+  records for the same person from being added to the database.
+* Privacy is a concern, as we are creating a database, presumably separate from the workspace,
+  which will need access control.
+  
+Estimated work: Unable to estimate without extensive research into uniquely identifying
+people globally and avoiding duplicate entries in a database
```

Comment:
It shouldn't be too difficult once one works out how to convert a human being into a data structure.

Reply:
Skynet is never around when you need it